src/EffectComposer.tsx

```diff
@@ -36,7 +36,7 @@ const EffectComposer = forwardRef(
       return [effectComposer, pass]
     }, [camera, gl, multisampling, props, scene])
 
-    useEffect(() => void composer.setSize(size.width, size.height), [composer, size])
+    useEffect(() => void composer.setSize(size.width * devicePixelRatio, size.height * devicePixelRatio), [composer, size, devicePixelRatio])
```

Comment:
Could you please put `window.` before acessing the ratio? nothing wrong here, just for better readability